### PR TITLE
Get rid of matlab implementation of split QP controller

### PR DESCRIPTION
**Pull request by rdeits**
We're not using it, and it's a time sink to maintain it, especially since it won't ever be fast enough to run on the robot. I've deleted all of the matlab code which duplicates functionality in QPCommon.cpp 

I've also deleted testSplitWalking.m from `examples/Atlas/test`. All it was doing was calling `runAtlasWalkingSplit` with `use_mex` set to 0, 1, and 2. Now we can just call `runAtlasWalkingSplit` once during unit testing.

### CODE REVIEW

**Comment by tkoolen** [comment]
I'm completely in favor of getting rid of all of this duplicated code, but I think @hongkai-dai was actually using this recently.

**Comment by hongkai-dai** [comment]
@tkoolen , I was using the MATLAB setupAndSolveQP, but then realized that the MATLAB version and C++ implementation have diverged already, that they do not give the same result. So I am using C++ setupAndSolveQP only.

File: examples/Atlas/+atlasParams/Base.m [hunk 1]
```diff
@@ -38,7 +38,7 @@
                               'w_qdd', 1e-3 * ones(r.getNumVelocities(), 1),...
                               'integrator', struct('gains', zeros(r.getNumPositions(), 1),...
                                                    'clamps', zeros(r.getNumPositions(), 1),...
-                                                   'eta', 0.005),...
+                                                   'eta', 0.0),...
                               'qdd_bounds', struct('min', -100*ones(r.getNumVelocities(), 1),...
```
Comment (RussTedrake) [inline examples/Atlas/+atlasParams/Base.m]: did you mean to sneak this in?

Comment (rdeits) [inline examples/Atlas/+atlasParams/Base.m]: There was an error in constructQPDataPointerMex in which `eta` was not being parsed (and was thus probabilistically zero). We're now correctly parsing `eta`, so by setting it to zero in the params, we match the prior behavior

**Comment by RussTedrake** [comment]
as discussed, robin will revert the small parameter changes just to be conservative.

File: examples/Atlas/+atlasParams/Walking.m [hunk 2]
```diff
@@ -11,8 +11,8 @@
       obj.whole_body.w_qdd(findPositionIndices(r, 'back')) = .0001;
       obj.whole_body.w_qdd(r.findPositionIndices('back_bkx')) = 0.1;
       obj.whole_body.damping_ratio = 0.5;
-      obj.body_motion(r.findLinkId('pelvis')).Kp = [nan; nan; 20; 20; 20; 20];
-      obj.body_motion(r.findLinkId('pelvis')).damping_ratio = 0.5;
+      obj.body_motion(r.findLinkId('pelvis')).Kp = [nan; nan; 150; 200; 200; 200];
+      obj.body_motion(r.findLinkId('pelvis')).damping_ratio = 0.6;
```
Comment (rdeits) [inline examples/Atlas/+atlasParams/Walking.m]: These changes were made to match the parameters we use in the DRC sim, which have been working really well, so I think we should keep this change

**Comment by rdeits** [comment]
Walking performance is somehow much worse on this branch. I'm bisecting now to figure out what went wrong and why I didn't see it before.

**Comment by rdeits** [comment]
The problem was that 860f9bb1c0298600dfd58f262c78dd90a572ad17 changed the nominal pelvis height from `0.84` to `0.82` AND that 5f3f919 resulted in the pelvis height being further reduced by approximately 0.5mm. That pushed us just beyond the edge of stable walking. I'm going to revert 860f9bb1c0298600dfd58f262c78dd90a572ad17 and bring the pelvis back to 0.84 by default for this branch.

**Comment by rdeits** [comment]
Ok, `navgoal` and `num_steps` have been restored, walking tracking looks good, and we now have an explicit check on CoM tracking. Should be good to go once the build servers have tested it.

**Pull request merged**